app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt

```diff
@@ -146,10 +150,17 @@ class NewProfileFragment : Fragment() {
                     vibSwitch = binding.vibSwitch.isChecked,
                     timeInstance = currentTime
                 )
-                profile.profileId = System.currentTimeMillis()
-                profileViewModel.insert(profile)
-                Navigation.findNavController(it)
-                    .navigate(NewProfileFragmentDirections.actionNewProfileFragmentToMainFragment())
+                if (binding.makeProfileFab.text == "Submit") {
+                    profile.profileId = System.currentTimeMillis()
+                    profileViewModel.insert(profile)
+                } else {
+                    profileListAdapter = ProfileListAdapter(profileViewModel, View(context), activity)
+                    profileListAdapter.removeWork(StoreSession.readLong(AppConstants.PROFILE_ID).toString())
+                    profile.profileId = StoreSession.readLong(AppConstants.PROFILE_ID)
```

Comment:
If profile ID is required at multiple places in a single file, then why not store it in a variable rather than accessing it multiple times from shared preferences?